### PR TITLE
fix Flaky-test: testBookieShouldTurnWritableFromReadOnly

### PR DESCRIPTION
**Pull request by wenbingshen**
### Motivation
Fixes issue #3422 

1.`waitForReadOnlyBookie` adds another `listener` thread to observe the node status of bookie, which may be out of sync with the triggering of node changes(`onClusterChanged`) in `EnsemblePlacementPolicy`.This `sequence` leads to flaky test. 

2.Expose bookie nodes in EnsemblePlacementPolicy to client.

3.Just change from watching zk to thread sleep.

We currently don't have any additional requirements for getting bookie nodes on the client side, so it's better to just use thread sleep here. 

### Changes
Change from `waitForReadOnlyBookie` and `waitForWritableBookie` to thread `sleep`.
3 seconds is enough for most.

### CODE REVIEW

**Comment by wenbingshen** [comment]
ping @hangc0276 @Shoothzj @dlg99 @eolivelli @nicoloboschi PTAL.

**Comment by wenbingshen** [comment]
rerun failure checks

File: bookkeeper-server/src/test/java/org/apache/bookkeeper/test/ReadOnlyBookieTest.java [hunk 1]
```diff
@@ -128,8 +129,10 @@ public void testBookieShouldTurnWritableFromReadOnly() throws Exception {
             // Expected
         }
 
-        bkc.waitForReadOnlyBookie(BookieImpl.getBookieId(confByIndex(1)))
-            .get(30, TimeUnit.SECONDS);
+        // waitForReadOnlyBookie adds another listener thread to observe the node status of bookie,
+        // which may be out of sync with the triggering of node changes in EnsemblePlacementPolicy.
+        // This sequence leads to flaky test. So change from watching zk to thread sleep.
+        Thread.sleep(3000);
```
Comment (hangc0276) [inline bookkeeper-server/src/test/java/org/apache/bookkeeper/test/ReadOnlyBookieTest.java:135]: Would you please use `Awayatility` instead of sleep ?

Comment (wenbingshen) [reply bookkeeper-server/src/test/java/org/apache/bookkeeper/test/ReadOnlyBookieTest.java]: sure. I have addressed your comments. PTAL.

**Comment by wenbingshen** [comment]
reopen trigger CI again

**Comment by wenbingshen** [comment]
rerun failure checks

**Review by hangc0276 — APPROVED** [review]

**Pull request merged**